superset/databases/api.py

```diff
@@ -238,6 +238,7 @@ def post(self) -> Response:
         except DatabaseInvalidError as ex:
             return self.response_422(message=ex.normalized_messages())
         except DatabaseConnectionFailedError as ex:
+            logger.exception("Database connection failed")
```

Comment:
Can we downgrade these from exception logging? Connection failures are an expected part of setting up new database connections and shouldn't wind up in exception tracking systems.

Reply:
If you want to understand the reasons for failures, it would also be a good idea to log the exception. However, beware: these exceptions frequently contain database credentials, which it would be bad to track.

Reply:
Good point on the level. Just a note, @willbarrett: `@logger.exception` logs the full traceback plus the message, so it would've helped figuring out the reason.

Reply:
`logger.warning("Database connection failed", exc_info=True)` also achieves the same. As the one currently running the error/exception logs portion of our biweekly alerts review, I would greatly appreciate if we reduced the amount of expected exceptions being logged at error and exception level. It would make my job significantly easier, as well as help reduce the strain on our quota

Reply:
I agree with @willbarrett concern, unfortunately some engines dbapi surface exceptions with sensitive information `logger.warning("Database connection failed", exc_info=True)` will actually log this info. we can lower it to debug, since on normal situations production systems are not set to this level.

Reply:
I'm now wondering if there's a better way to get the info that @betodealmeida and @hughhhh are looking for - would you be able to describe the questions that you're trying to answer with this logging? Then we can support you better in coming to a safe solution.

Reply:
We want to answer the following:
What db engines are failing the most?
What are the test connection and db connection exception error for these db engines?

If we can get this information with other data let us know

Reply:
What DB engines are failing the most?
I think you could parse the submitted SQLAlchemy URL to pull out the `scheme` portion of the URL and log that on failure. This would avoid the risk of capturing secure credentials.

What are the the test connection and DB connection exceptions for these engines?
Would it be sufficient to log the exception class from the engine without the message? The class will give you the category of error without including any of the potentially sensitive details. If this would work, you could more easily perform this logging from the command rather than in the endpoint.

Reply:
This is a great use case for statsd!